borg/testsuite/repository.py

```diff
@@ -201,11 +201,14 @@ def test_ignores_commit_tag_in_data(self):
 
 
 class RepositoryAppendOnlyTestCase(RepositoryTestCaseBase):
+    def open(self, create=False):
+        return Repository(os.path.join(self.tmppath, 'repository'), create=create, append_only=True)
+
     def test_destroy_append_only(self):
         # Can't destroy append only repo (via the API)
-        self.repository.append_only = True
         with self.assert_raises(ValueError):
             self.repository.destroy()
+        assert self.repository.append_only == True
 
```

Comment:
I don't quite see how this would test the new code. It would need a subclass analogous to the other subclasses to do that, like 

```
 class RemoteRepositoryAppendOnlyTestCase(RepositoryAppendOnlyTestCase):
 
     def open(self, create=False):
         return RemoteRepository(Location('__testsuite__:' + os.path.join(self.tmppath, 'repository')), create=create)
 ```